internal/pkg/handler/resourcehandler.go

```diff
@@ -0,0 +1,58 @@
+package handler
+
+import (
+	"github.com/sirupsen/logrus"
+	"k8s.io/api/core/v1"
+)
+
+// ResourceHandler handles the creation and update of resources
+type ResourceHandler interface {
+	Handle() error
+}
+
+// ResourceCreatedHandler contains new objects
+type ResourceCreatedHandler struct {
```

Comment:
Refactor these created and updated handlers to separate files. created-handler.go, updated-handler.go